Frontend refactoring

furinvader opened the pull request:
**Module Loading**
The Intention is to make components (`.html` and `.js` files) being loaded without any extra configuration. Just put them under `frontend/app/components`.

**Code Style**
Also, a new way of naming component files is introduced.
Some examples: `my-thing.controller.js`, `hamburgers.service.js` ... for tests `mamamia.test.js`, `some-e2e.spec.js`

**Testing**
For better overview, put test files in the same location as their respective components.

furinvader committed:
Simplify component handling

[ci skip]

furinvader committed:
Refactor HALAPIResource service

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://furinvader-features-refactoring.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://furinvader-features-refactoring.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

furinvader committed:
Allow tests/specs to be located under components/

This makes it possible to have tests in the same
 location as the components they belong to.

furinvader committed:
Fix error that prevented protractor from starting

[ci skip]
